src/scripts/mutual_checker.js

```diff
@@ -50,72 +74,103 @@ const addIcons = function (postElements) {
-        .catch(() => Promise.resolve(false));
-    }
-
-    const isMutual = await mutuals[blogName];
+    const isMutual = await getIsFollowingYou(blogName);
     if (isMutual) {
       postElement.classList.add(mutualsClass);
       const iconTarget = getPopoverWrapper(postAttribution) ?? postAttribution;
-      iconTarget?.before(icon.cloneNode(true));
+      iconTarget?.before(createIcon(blogName));
     } else if (showOnlyMutuals) {
       getTimelineItemWrapper(postElement)?.setAttribute(hiddenAttribute, '');
     }
   });
 };
 
+const addBlogCardIcons = blogCardLinks =>
+  blogCardLinks.forEach(async blogCardLink => {
+    const blogName = blogCardLink.querySelector(keyToCss('blogLinkShort'))?.textContent || blogCardLink?.textContent;
+    if (!blogName) return;
+
+    const followingBlog = await getIsFollowing(blogName, blogCardLink);
+    if (!followingBlog) return;
+
+    const isMutual = await getIsFollowingYou(blogName);
+    if (isMutual) {
+      blogCardLink.before(createIcon(blogName, blogCardLink.style.color));
+    }
+  });
+
+const getIsFollowing = async (blogName, element) => {
+  const blog = await blogData(element) ?? (await timelineObject(element))?.blog;
+
+  if (following[blogName] === undefined) {
+    if (blogName === blog?.name) {
+      following[blogName] = Promise.resolve(blog.followed && !blog.isMember);
+    } else {
+      following[blogName] = apiFetch(`/v2/blog/${blogName}/info`)
+        .then(({ response: { blog: { followed } } }) => followed)
+        .catch(() => Promise.resolve(false));
+    }
+  }
+  return following[blogName];
+};
+
+const getIsFollowingYou = (blogName) => {
+  if (followingYou[blogName] === undefined) {
+    followingYou[blogName] = apiFetch(`/v2/blog/${primaryBlogName}/followed_by`, { queryParams: { query: blogName } })
+      .then(({ response: { followedBy } }) => followedBy)
+      .catch(() => Promise.resolve(false));
+  }
+  return followingYou[blogName];
+};
+
 export const main = async function () {
   if (primaryBlogName === undefined) return;
+  document.documentElement.append(styleElement);
 
   ({ showOnlyMutuals, showOnlyMutualNotifications } = await getPreferences('mutual_checker'));
   following[primaryBlogName] = Promise.resolve(false);
 
+  onNewPosts.addListener(addIcons);
+  pageModifications.register(`${keyToCss('blogCard')} ${keyToCss('blogCardBlogLink')} > a`, addBlogCardIcons);
+
+  if (showOnlyMutualNotifications) {
+    document.documentElement.append(onlyMutualsStyleElement);
+    onNewNotifications.addListener(processNotifications);
+  }
+};
+
+const createIcon = (blogName, color) => {
   const today = new Date();
   const aprilFools = (today.getMonth() === 3 && today.getDate() === 1);
 
-  icon = dom('svg', {
+  const icon = dom('svg', {
     xmlns: 'http://www.w3.org/2000/svg',
     class: mutualIconClass,
     viewBox: '0 0 1000 1000',
-    fill: aprilFools ? '#00b8ff' : 'rgb(var(--black))'
+    fill: color || 'rgb(var(--black))'
```

Comment:
1. Do we ever expect `color` to be an empty string here?
2. I feel like we can keep the `aprilFools` ternary, and just chain it with the fallback

```suggestion
    fill: aprilFools ? '#00b8ff' : (color ?? 'rgb(var(--black))')
```

Reply:
I think what I was supposed to do here was set a default argument for the function, tbh.